### PR TITLE
Fix: Redirect when canceling the deleltion of comment; Closes #1497

### LINKED ISSUE #1497: Delete Comment -> Cancel button does nothing
Goes to the delete comment (same) page...

image

Reproduce:

Create an announcement, then add a comment below. Click the delete icon to delete the comment to go somewhere like this: https://hackerspace.bytedeck.com/comments/274249/delete/

### PR DESCRIPTION
**Pull request by 9921622**
Please ensure you are familiar with our Pull Request Description expectations described here: https://www.pullrequest.com/blog/writing-a-great-pull-request-description/
### What?
Stated in #1497, there is an issue with the cancel button when deleting a comment (via comment confirm delete.)
This PR fixes that by redirecting the comment to its comment.path URL, or the source link of where the user commented. 

### Why?
This issue is an unexpected behaviour as the cancel button implies that it cancels the deleltion of a comment. In reality, it does nothing.

### How?
By changing the "comment" variable to "object" in src/comments/templates/comments/confirm_delete.html. As "comment" is not referenced at all in the context but "object" is. 
From:
```
{{comment.path}}
```
To:
```
{{object.path}}
```

### Testing?
Manually clicking cancel with an admin account and visually confirming if it redirected to the announcement page.

Also created an automated test inside comments test apps, where (using bs4) checks if the comments path is inside the cancel button.
ie.
``` <a href="COMMENT PATH" role="button" class="btn btn-info">Cancel</a> ```

### Screenshots (if front end is affected)
Before pressing cancel:
image
After pressing cancel:
image

### Anything Else?
I've also noticed this behavior also applies to comments for quests submissions. This pr should also fix it. Tested using the same method mentioned above

### Review request
@tylerecouture

### CODE REVIEW

File: src/comments/templates/comments/confirm_delete.html [hunk 1]
```diff
@@ -11,7 +11,7 @@
     <p>Date Posted: {{object.timestamp}}</p>
     <p>Comment: {{object.text|safe}}</p>
   </div>
-  <a href="{{comment.path}}" role="button" class="btn btn-info">Cancel</a>
+  <a href="{% url 'announcements:list' %}" role="button" class="btn btn-info">Cancel</a>
```
[the remark this answers is not in the document]
Comment (tylerecouture) [reply src/comments/templates/comments/confirm_delete.html]: @coderabbitai yes

[the remark this answers is not in the document]
Comment (tylerecouture) [reply src/comments/templates/comments/confirm_delete.html]: @coderabbitai Nah, this is just testing the view.  It doesn't actually simulate clicking the button. For that we would need a browser emulator.  The view is already being tested elsewhere I assume

[the remark this answers is not in the document]
Comment (tylerecouture) [reply src/comments/templates/comments/confirm_delete.html]: @9921622 Can you try this test with the OLD code and see if it fails?  and then the new code, does it pass?  If that's the case, then it's actually a useful test.

[the remark this answers is not in the document]
Comment (9921622) [reply src/comments/templates/comments/confirm_delete.html]: do we even have selenium? it seems pretty costly to run just for testing. iirc we have beautiful soup and I can prob just assert the object.path exist within the button using that

[the remark this answers is not in the document]
Comment (9921622) [reply src/comments/templates/comments/confirm_delete.html]: @tylerecouture forgot to tag you

[the remark this answers is not in the document]
Comment (tylerecouture) [reply src/comments/templates/comments/confirm_delete.html]: @9921622 No we haven't implemented selenium tests.  Your suggesting sounds good.

[the remark this answers is not in the document]
Comment (9921622) [reply src/comments/templates/comments/confirm_delete.html]: Ok  i've created bs4 tests for the cancel button

File: src/comments/tests/test_views.py [hunk 2]
```diff
@@ -95,3 +96,18 @@
...
+
+        # get confirm delete content html
+        response = self.client.get(reverse('comments:delete', args=[self.comment.id]))
+        self.assertContains(response, 'Cancel')
+
+        soup = BeautifulSoup(response.content.decode('utf-8'), features='html.parser')
+
+        # find the Cancel Button
+        tag = soup.find('a', href=self.announcement.get_absolute_url(), role='button', text='Cancel')
+        self.assertIsNotNone(tag)
```
[the remark this answers is not in the document]
Comment (tylerecouture) [reply src/comments/tests/test_views.py:113]: @coderabbitai your first docstring was better, but can you put it on two lines?

[the remark this answers is not in the document]
Comment (tylerecouture) [reply src/comments/tests/test_views.py:113]: @coderabbitai this is perfect, can you make it a committable suggestion?

**Review by tylerecouture — APPROVED** [review]

**Pull request merged**